File: Easy/Reach the Nth point.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
	public:
	
		int nthPoint(int n)
		{
		    // Code here
		    if (n<=2) return n;
    
    	    int m = pow(10,9)+7;
		    int last1 = 2;
		    int last2 = 1;
		    
		    int ans = 0;
		    for (int i=2; i<n; i++)
		    {
		       ans = (last1 + last2) %m;
		       last2 = last1 %m;
		       last1 = ans;
		    }
		    
		    return ans;
		    
		}
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		int n;
		cin >> n;
		Solution ob;
		int ans  = ob.nthPoint(n);
		cout << ans <<"\n";
	}
	return 0;
}
// } Driver Code Ends

File: Easy/Reach the Nth point.py
#User function Template for python3

class Solution:
	def nthPoint(self, n):
		# Code here
		m = 1000000007
		
		if n <= 2:
		    return n
		    
        a = 1
        b = 1
        fibo = 2
        
        for i in range(1, n):
            fibo = (a + b) % m
            a = b
            b = fibo
        
        return fibo
		


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import sys
sys.setrecursionlimit(10**6)

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n = int(input())
		ob = Solution();
		ans = ob.nthPoint(n)
		print(ans)
# } Driver Code Ends